content/blog/2023-03-23-when-a-date-breaks-booting-the-kernel.md

```diff
@@ -0,0 +1,213 @@
+Title: When a date breaks booting the kernel
+Date: 2023-03-23 00:00:00
+Authors: Benjamin Gray
+Category: Development
+Tags: linux
+
+## The setup
+
+I've recently been working on internal CI infrastructure for testing kernels before sending them to the mailing list. As part of this effort, I became interested in reproducible builds. Minimising the changing parts outside of the source tree itself could improve consistency and ccache hits, which is great for trying to make the CI faster and more reproducible.
+
+As part of this effort, I came across the `KBUILD_BUILD_TIMESTAMP` environment variable. This variable is used to set the kernel timestamp, which is primarily just baked into the version string for any users who want to know when their kernel was built. That's mostly irrelevant for our work, so an easy `KBUILD_BUILD_TIMESTAMP=0` later and... it still uses the current date.
```

Comment:
> '... timestamp, which is primarily just baked into the version string for any users who want to know when their kernel was built.'

Replace with '... timestamp, baked into the version string primarily for any users ...', makes the object of 'primarily' easier to ascertain.